asdf/tests/helpers.py

```diff
@@ -233,6 +233,15 @@ def assert_roundtrip_tree(tree, tmpdir, *, asdf_check_func=None,
         finally:
             server.finalize()
 
+    # Now don't be lazy and check that nothing breaks
+    with io.BytesIO() as buff:
+        AsdfFile(tree, extensions=extensions, **init_options).write_to(buff, **write_options)
+        buff.seek(0)
+        ff = AsdfFile.open(buff, extensions=extensions, copy_arrays=True, lazy_load=False)
```

Comment:
I wonder if we also want a test with `copy_arrays=False`? Unless you think that `lazy_load=False` always implies `copy_arrays=True`, in which case we may need to make a change to the top-level API.

Reply:
Yeah, I will add the test.